### PR TITLE
Update coursier to 2.1.0-M2

### PR DESCRIPTION
**Pull request by alexarchambault**
Compared to `2.0.16-200-ge888c6dea`, this version fixes issues with snapshot artifacts, and also adds new APIs (`ArchiveCache`, that can be useful from Mill tasks).

### CODE REVIEW

File: build.sc [hunk 1]
```diff
@@ -92,9 +92,8 @@ object Deps {
   )
   val asciidoctorj = ivy"org.asciidoctor:asciidoctorj:2.4.3"
   val bloopConfig = ivy"ch.epfl.scala::bloop-config:1.4.13"
-  val coursier = ivy"io.get-coursier::coursier:2.0.16-200-ge888c6dea"
+  val coursier = ivy"io.get-coursier::coursier:2.1.0-M2"
   val coursierReducedDeps = coursier.exclude(
-    "com.lihaoyi" -> "utest",
```
Comment (lefou) [inline build.sc:97]: Is this no longer in the transitive dependency set, or is it mandatory now?

Comment (alexarchambault) [reply build.sc:97]: It's not a dependency any more, it had ended up there by mistake…

Comment (lefou) [reply build.sc:97]: Ok. Nice you fixed it.

**Review by lefou — APPROVED** [review]
LGTM, thanks!

**Pull request merged**